src/hazelcore/CursorInfo_Exp.re

```diff
@@ -159,7 +159,8 @@ and get_outer_zrules_from_zrule =
 };
 
 let rec cursor_on_outer_expr:
-  ZExp.zoperand => option((ErrStatus.t, VarErrStatus.t)) =
+  ZExp.zoperand =>
+  option((ErrStatus.t, VarErrStatus.t, option(list(HTyp.t)))) =
```

Comment:
the three elements of this tuple are exclusive right? should use a sum type here in that case rather than a product.